src/e2e/specs/dashboard.spec.ts

```diff
@@ -127,6 +133,94 @@ test.describe(`${process.env.E2E_TEST_ENV} - Breaches Dashboard - Headers`, () =
     await dashboardPage.popupCloseButton.click();
     await expect(dashboardPage.aboutFixedExposuresPopup).toBeHidden();
   });
+
+  test("Verify that the Apps and Services header options work correctly.", async ({
+    dashboardPage,
+    page,
+  }) => {
+    // link to testrail
+    test.info().annotations.push({
+      type: "testrail",
+      description:
+        "https://testrail.stage.mozaws.net/index.php?/cases/view/2463569",
+    });
+
+    await dashboardPage.fixedTab.click();
+    expect(page.url()).toMatch(/.*dashboard\/fixed\/?/);
+    await dashboardPage.actionNeededTab.click();
+    expect(page.url()).toMatch(/.*dashboard\/action-needed\/?/);
+
+    //apps and services button check
+    const clickOnLinkAndGoBack = async (
+      aTag: Locator,
+      host: string | RegExp = /.*/,
+      path: string | RegExp = /.*/,
+    ) => {
+      await expect(dashboardPage.appsAndServices).toBeVisible();
+      await dashboardPage.appsAndServices.click();
+      await expect(dashboardPage.appsAndServicesMenu).toBeVisible();
+      await clickOnATagCheckDomain(aTag, host, path, page);
+    };
+
+    await clickOnLinkAndGoBack(
+      dashboardPage.servicesVpn,
+      "www.mozilla.org",
+      /.*\/products\/vpn\/?.*/,
+    );
+    await clickOnLinkAndGoBack(
+      dashboardPage.servicesRelay,
+      "relay.firefox.com",
+    );
+    await clickOnLinkAndGoBack(
+      dashboardPage.servicesPocket,
+      /getpocket\.com|apps\.apple\.com|app\.adjust\.com/,
+      /.*(\/pocket-and-firefox\/?).*|.*about.*|.*pocket-stay-informed.*/,
```

Comment:
Similarly here... it's getting a bit confusing since I'm trying to figure out which paths belong to which hosts between these two regexps. 🤔

Reply:
Do you think there is a better way to do this if there are so many hosts and so many paths that are thus possible?